app/lib/features/activities/widgets/space_activities_section/item_widgets/activity_membership_container_widget.dart

```diff
@@ -0,0 +1,190 @@
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/providers/common_providers.dart';
+import 'package:acter/common/widgets/acter_icon_picker/utils.dart';
+import 'package:acter/features/comments/widgets/time_ago_widget.dart';
+import 'package:acter/l10n/generated/l10n.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+/// A widget that displays membership change events in a space's activity feed.
+/// 
+/// This widget shows information about various membership changes like:
+/// - Users joining/leaving a space
+/// - Invitations being sent/accepted/rejected
+/// - Users being banned/unbanned
+/// - Knock requests and their status
+class ActivityMembershipItemWidget extends ConsumerWidget {
+  /// The activity containing membership change information
+  final Activity activity;
+
+  const ActivityMembershipItemWidget({
+    super.key,
+    required this.activity,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final activityObject = activity.object();
+    final originServerTs = activity.originServerTs();
+    
+    return Card(
+      margin: const EdgeInsets.symmetric(horizontal: 0, vertical: 8),
+      child: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 12),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.start,
+          crossAxisAlignment: CrossAxisAlignment.end,
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            _buildActionInfoUI(context, ref, activityObject),
+            const SizedBox(height: 6),
+            _buildUserInfoUI(context, ref),
+            TimeAgoWidget(originServerTs: originServerTs),
+          ],
+        ),
+      ),
+    );
+  }
+
+  /// Builds the UI section showing the membership change action and its details
+  Widget _buildActionInfoUI(BuildContext context, WidgetRef ref, ActivityObject? activityObject) {
+    final actionTitleStyle = Theme.of(context).textTheme.labelMedium;
+    final membershipInfo = _getMembershipInfo(context, ref, activityObject);
+    final actionIconColor = Theme.of(context).colorScheme.onSurface;
```

Comment:
just a minor re-ordering for better legibility
```suggestion
    final membershipInfo = _getMembershipInfo(context, ref, activityObject);
    
    final actionTitleStyle = Theme.of(context).textTheme.labelMedium;
    final actionIconColor = Theme.of(context).colorScheme.onSurface;
```

Reply:
Addressed in this [commit](https://github.com/acterglobal/a3/pull/2746/commits/c251d510301ca25505a93286199ea48604860f31)